File: python-server/app/utils/time_utils.py
import datetime
import zoneinfo

from ..config import TIMEZONE


# Convert a Timestamp into Minutes since Midnight.
def minute_of_day(epoch_ms: int) -> int:
    dt = datetime.datetime.fromtimestamp(
        epoch_ms / 1000,
        tz=zoneinfo.ZoneInfo(TIMEZONE),
    )
    return dt.hour * 60 + dt.minute


# Convert a Timestamp into the Weekday index (Monday=0 ... Sunday=6).
def day_of_week(epoch_ms: int) -> int:
    dt = datetime.datetime.fromtimestamp(
        epoch_ms / 1000,
        tz=zoneinfo.ZoneInfo(TIMEZONE),
    )
    return dt.weekday()

File: node-server/src/utils/time.utils.ts
const TIMEZONE = process.env.TIMEZONE || "Asia/Kolkata";


// Convert a Timestamp into local Minute-of-Day and Day-of-Week.
export function localTimeParts(ts: number): { minuteOfDay: number; dayOfWeek: number } {

  const fmt = new Intl.DateTimeFormat("en-US", {
    timeZone: TIMEZONE,
    hour12:   false,
    hour:    "2-digit",
    minute:  "2-digit",
    weekday: "short",
  });
  
  const parts  = fmt.formatToParts(new Date(ts));
  const get    = (type: string) => parts.find((p) => p.type === type)?.value ?? "0";
  const hour   = parseInt(get("hour"), 10) % 24;
  const minute = parseInt(get("minute"), 10);

  const weekdayMap: Record<string, number> = { Mon: 0, Tue: 1, Wed: 2, Thu: 3, Fri: 4, Sat: 5, Sun: 6 };
  return { minuteOfDay: hour * 60 + minute, dayOfWeek: weekdayMap[get("weekday")] ?? 0 };
}